easy_profiler_core/include/easy/details/easy_compiler_support.h

```diff
@@ -191,8 +193,13 @@ static_assert(false, "EasyProfiler is not configured for using your compiler typ
 #endif
 
 #ifndef EASY_LOCAL_STATIC_PTR
-# define EASY_LOCAL_STATIC_PTR(VarType, VarName, VarInitializer) static VarType VarName = VarInitializer
-# define EASY_MAGIC_STATIC_AVAILABLE
+# if defined(_MSC_VER) && EASY_DISABLE_THREAD_SAFE_INIT != 0
+#  define EASY_LOCAL_STATIC_PTR(VarType, VarName, VarInitializer) thread_local static VarType VarName = VarInitializer
+#  define EASY_MAGIC_STATIC_AVAILABLE
```

Comment:
EASY_MAGIC_STATIC_AVAILABLE should not be defined if thread-safe static init is disabled